lib/mayaUsd/fileio/utils/writeUtil.h

```diff
@@ -301,6 +302,34 @@ struct UsdMayaWriteUtil
 
     /// \}
 
+    /// Sets the value of \p attr to \p value at \p time with value
+    /// compression. When this method is used to write attribute values,
+    /// any redundant authoring of the default value or of time-samples
+    /// are avoided (by using the utility class UsdUtilsSparseValueWriter).
+    template <typename T>
+    static bool SetAttribute( const UsdAttribute& attr,
+                              const T& value,
+                              UsdUtilsSparseValueWriter& valueWriter,
+                              const UsdTimeCode time = UsdTimeCode::Default()) 
+    {
+        VtValue val(value);
+        return valueWriter.SetAttribute(attr, &val, time);
+    }
+
+    /// \overload
+    /// This overload takes the value by pointer and hence avoids a copy
+    /// of the value.
+    /// However, it swaps out the value held in \p value for efficiency,
+    /// leaving it in default-constructed state (value-initialized).
+    template <typename T>
+    static bool SetAttribute( const UsdAttribute& attr,
+                              T* value,
+                              UsdUtilsSparseValueWriter& valueWriter,
+                              const UsdTimeCode time = UsdTimeCode::Default()) 
+    {
+        return valueWriter.SetAttribute(attr, VtValue::Take(*value), time);
+    }
+
```

Comment:
@mattyjams you have probably seen by now but _SetAttribute eventually is removed:

https://github.com/Autodesk/maya-usd/pull/428#pullrequestreview-403932691

I didn't remove it in this PR to avoid making lots of additional changes.